packages/router/history.ts

```diff
@@ -694,6 +694,7 @@ function getUrlBasedHistory(
     // pre-encode them since they might be part of a matching splat param from
     // an ancestor route
     href = href.replace(/ $/, "%20");
+    href = typeof to === "string" ? href.replace(/#/, "%23") : href
```

Comment:
This will always be a string. This test shouldn't be needed.
```suggestion
    href = href.replace(/#/, "%23")
```

Reply:
<img width="953" alt="image" src="https://github.com/user-attachments/assets/d4cd90f1-67a8-440e-ad85-74c456757c58">
In this test, when executing fireEvent.click(screen.getByText('Link to path')), to is not a string; it is an object and does not need to be replaced.

Reply:
<img width="1335" alt="image" src="https://github.com/user-attachments/assets/a4741ea3-ce8b-4e6b-8d33-a922ef6a3f86">
This test fails when href = href.replace(/#/, '%23') is executed.